### DISCUSSION TITLE
Update Typescript on `MantineProvider`

### DISCUSSION THREAD

**Discussion by maclementED** [mantinedev/mantine#3072] [Feature requests]
I could be great to improve the typescript interface of the component `MantineProvider`. 

### Problem
Right now, the `components` property does not indicate if the component is valid:
![image](https://user-images.githubusercontent.com/67877770/204334747-0162ad2f-7641-4ec1-a0c8-29ccbfe00a6e.png)

Furthermore, the `defaultProps` property is not linked to his component props. When adding properties to the Button component, it could be nice to see the have `ButtonProps` interface instead. 
![image](https://user-images.githubusercontent.com/67877770/204333795-6ae997bb-6779-464d-8840-50856871aa18.png)

### Solution
A simple solution would be to update the current `MantineThemeComponents` interface to something similar to this: 
![image](https://user-images.githubusercontent.com/67877770/204336529-f08683ec-dd06-4a91-a11e-7e38c7946114.png)
<sub>There is most likely a better solution</sub>

**Comment by rtivital** [comment]
- Adding all types to MantineProvider off 200+ components will significantly degrade the performance
- Correct types cannot be added for polymorphic components, as their real type is calculated based on `component` prop
- Current approach allows any third-party libraries to use MantineProvider features seamlessly – for example, with strict types it won't be easy to add new component that is not a part of the library
